### DISCUSSION TITLE
Is static linking required? (Expo)

### DISCUSSION THREAD

**Discussion by dylancom** [callstack/react-native-bottom-tabs#219] [Q&A] [answered]
The docs now mention that you need to enable static linking for iOS.
But in for e.g. the advised example app: https://github.com/EvanBacon/expo-router-react-native-bottom-tabs
I don't see any static linking being set at all.
Any info on why this is needed or in some cases maybe not?

**Comment by okwasniewski** [comment, marked as answer]
Hey, yes - static linking is required. 

The example app is outdated, I need to update it to a newer one. In 0.5.0 (https://github.com/okwasniewski/react-native-bottom-tabs/releases/tag/v0.5.0) I've introduced another native dependency to handle SVG support. This dependency doesn't define modules (and all of the cocoapods quirks that go with it) require us to enable static linking. 

There is a workaround you can do if you really don't want to enable static linking: https://github.com/okwasniewski/cocoapods-swift-modular-headers

You can create a simple Expo config plugin to inject this plugin into your Podfile and this will solve the issue with modular headers. Expo does this in their autolinking but only for Expo Modules.

**Comment by elsa17z** [comment]
I have figured out without either adding the Cocoapods plugin nor using static linking. 

You have to install expo-build-properties and specify the 2 dependencies that depend on it as extraPods, then the modular_headers would be automatically applied and override the default settings. 

@dylancom 
@okwasniewski Would it be possible to add this to the docs as well? I figured this would be way easier than any other way to do it. I was constantly getting errors when it comes to static linking. 

ios: {
            extraPods: [
              { name: "SDWebImage", modular_headers: true }, // Work around for not enabling static framework, required for react-native-bottom-tabs
              { name: "SDWebImageSVGCoder", modular_headers: true }, // See https://github.com/callstackincubator/react-native-bottom-tabs/blob/1b0ed94579bb234afc047a0bae7f2463cb125de0/example/ios/Podfile
            ],
          },
